recipes/leptonica/all/conanfile.py

```diff
@@ -139,47 +137,66 @@ def _patch_sources(self):
+            if Version(self.version) >= "1.83.0":
+                replace_in_file(self, cmake_configure, "if(GIF_FOUND)", "if(0)")
+            else:
+                replace_in_file(self, cmake_configure, "if (GIF_FOUND)", "if(0)")
         ## libjpeg
         replace_in_file(self, cmakelists_src, "${JPEG_LIBRARIES}", "JPEG::JPEG")
         if not self.options.with_jpeg:
             replace_in_file(self, cmakelists_src, "if (JPEG_LIBRARIES)", "if(0)")
-            replace_in_file(self, cmake_configure, "if (JPEG_FOUND)", "if(0)")
+            if Version(self.version) >= "1.83.0":
+                replace_in_file(self, cmake_configure, "if(JPEG_FOUND)", "if(0)")
+            else:
+                replace_in_file(self, cmake_configure, "if (JPEG_FOUND)", "if(0)")
         ## libpng
         replace_in_file(self, cmakelists_src, "${PNG_LIBRARIES}", "PNG::PNG")
         if not self.options.with_png:
             replace_in_file(self, cmakelists_src, "if (PNG_LIBRARIES)", "if(0)")
-            replace_in_file(self, cmake_configure, "if (PNG_FOUND)", "if(0)")
+            if Version(self.version) >= "1.83.0":
+                replace_in_file(self, cmake_configure, "if(PNG_FOUND)", "if(0)")
+            else:
+                replace_in_file(self, cmake_configure, "if (PNG_FOUND)", "if(0)")
         ## libtiff
         replace_in_file(self, cmakelists_src, "${TIFF_LIBRARIES}", "TIFF::TIFF")
         if not self.options.with_tiff:
             replace_in_file(self, cmakelists_src, "if (TIFF_LIBRARIES)", "if(0)")
-            replace_in_file(self, cmake_configure, "if (TIFF_FOUND)", "if(0)")
+            if Version(self.version) >= "1.83.0":
+                replace_in_file(self, cmake_configure, "if(TIFF_FOUND)", "if(0)")
+            else:
+                replace_in_file(self, cmake_configure, "if (TIFF_FOUND)", "if(0)")
         ## We have to be more aggressive with dependencies found with pkgconfig
         ## Injection of libdirs is ensured by conan_basic_setup()
         ## openjpeg
-        replace_in_file(self, cmakelists, "if(NOT JP2K)", "if(0)")
-        replace_in_file(self, cmakelists_src,
-                              "if (JP2K_FOUND)",
-                              "if (JP2K_FOUND)\n"
-                              "target_link_directories(leptonica PRIVATE ${JP2K_LIBRARY_DIRS})\n"
-                              "target_compile_definitions(leptonica PRIVATE ${JP2K_CFLAGS_OTHER})")
-        if not self.options.with_openjpeg:
-            replace_in_file(self, cmakelists_src, "if (JP2K_FOUND)", "if(0)")
-            replace_in_file(self, cmake_configure, "if (JP2K_FOUND)", "if(0)")
+        replace_in_file(self, cmakelists_src, "${JP2K_LIBRARIES}", "openjp2")
+        if Version(self.version) < "1.83.0":
+            # pkgconfig is prefered to CMake. Disable pkgconfig so only CMake is used
+            replace_in_file(self, cmakelists, "pkg_check_modules(JP2K libopenjp2>=2.0 QUIET)", "")
```

Comment:
Surprising. This recipe has pkgconfig in build requirements when openjpeg option is enabled, and pkgconfig files are generated by PkgConfigDeps, so why would it fail without this patch? I can try to test but AFAIR it was working fine.

Reply:
I've tested and previous patch works like a charm, openjpeg is always found & injected

Reply:
Yes, my bad. Previously openjpeg was located using pkgconfig. So hardcoded openjpeg's version paths do not affect already found openjpeg.

Why I have moved to cmake-only openjpeg is to fix MSVC build (I don't have a clue why it is failed). 1.83.0 and pre-1.83.0 Leptonica have different resolution order for openjpeg: pkg-config then cmake vs cmake vs pkg-config. I think it is easier for us just to use CMake. How do you think?